dluhc-component-library/src/components/maps/baseMap.tsx

```diff
@@ -0,0 +1,45 @@
+import { View } from "ol";
+import TileLayer from "ol/layer/Tile";
+import { useGeographic } from "ol/proj";
+import { OSM } from "ol/source";
+import { CSSProperties } from "preact/compat";
+import { useEffect, useMemo } from "preact/hooks";
+import "../../../node_modules/ol/ol.css";
+import { useMap } from "../../contexts/mapContext";
+
+interface BaseMapProps {
+  mapId: string;
+  lat?: number;
+  lng?: number;
+  zoom?: number;
+  className?: string;
+  style?: CSSProperties;
+}
+
+function BaseMap({
+  mapId,
+  lat = 0,
+  lng = 0,
+  zoom = 0,
+  className,
+  style,
+}: BaseMapProps) {
+  const map = useMap();
+  const props = useMemo(() => ({ className, style }), [className, style]);
+
+  useEffect(() => {
+    useGeographic();
+    map.setLayers([new TileLayer({ source: new OSM() })]);
+    map.setView(
+      new View({
+        center: [lng, lat],
+        zoom,
+      }),
+    );
+    map.setTarget(mapId);
+  }, [lng, lat, mapId, zoom]);
```

Comment:
map should also be in the array. Probably need to get eslint set up properly to catch this for us

Reply:
Agreed, the lint rules would be good to have